### PR TITLE
Fix cyclus compilation without cython

### PR DESCRIPTION
**Pull request by bam241**
this should allow user to compile cycles without cython

### CODE REVIEW

**Comment by bam241** [comment]
I still have an issue with the unit_test

any ideas ?

File: src/pyhooks.cc [hunk 1]
```diff
@@ -65,6 +65,8 @@ std::string JsonToPy(std::string infile) { return CyclusJsonToPy(infile); };
 }  // namespace toolkit
 }  // namespace cyclus
 #else   // else CYCLUS_WITH_PYTHON
+#include "error.h"
```
Comment (scopatz) [inline src/pyhooks.cc]: Nice catch

File: CMakeLists.txt [hunk 2]
```diff
@@ -323,11 +323,11 @@ IF(NOT CYCLUS_DOC_ONLY)
...
         "${HDF5_INCLUDE_DIRS}"
         "${Boost_INCLUDE_DIR}"
-        "${COIN_INCLUDE_DIRS}")
+        "${COIN_INCLUDE_DIRS}"
+        "${PYTHON_INCLUDE_DIRS}" )
 
     if(Cython_FOUND)
-        INCLUDE_DIRECTORIES("${PYTHON_INCLUDE_DIRS}"
-            "${NUMPY_INCLUDE_DIRS}")
+        INCLUDE_DIRECTORIES("${NUMPY_INCLUDE_DIRS}")
```
Comment (scopatz) [inline CMakeLists.txt]: I don't think that the changes to this file are correct.  It doesn't make sense to include the python dirs if we don't have cython can't build the Python bindings

Comment (bam241) [inline CMakeLists.txt]: without this I got:
```
/Users/mouginot/work/app/cyclus/cyclus/src/eventhooks.cc:4:10: fatal error: 'Python.h' file not found
#include "Python.h"
         ^
1 error generated.
make[2]: *** [src/CMakeFiles/cyclus.dir/eventhooks.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [src/CMakeFiles/cyclus.dir/all] Error 2
make: *** [all] Error 2
Traceback (most recent call last):
  File "install.py", line 208, in <module>
    main()
  File "install.py", line 205, in main
    install_cyclus(args)
  File "install.py", line 104, in install_cyclus
    shell=(os.name == 'nt'))
  File "/opt/local/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/subprocess.py", line 271, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['make', '-j4']' returned non-zero exit status 2
```

Comment (scopatz) [inline CMakeLists.txt]: I suspect that this wasn't from a clean build...

Comment (bam241) [inline CMakeLists.txt]: @scopatz I tried, both:
`python3 install.py --clean-build -j4`
`rm -rf build; python3 install.py --clean-build -j4`
with the same results.

Comment (scopatz) [inline CMakeLists.txt]: Did you do a `git clean -fx`?  You seem to have a bunch of ignored files hanging around the repo.

Comment (bam241) [inline CMakeLists.txt]: nice catch @scopatz ! 

I put it back at the right place.

Comment (scopatz) [inline CMakeLists.txt]: Great!

**Comment by scopatz** [comment]
This should be good to go in once CI is done.

**Comment by scopatz** [comment]
Also, please merge from develop

**Comment by scopatz** [comment]
Yup! Thanks!

**Pull request merged**